### ISSUE TITLE
Debug build problem in Ubuntu

### ISSUE THREAD

**Issue by brpetrucci** [revbayes/revbayes#106]
In attempting to set up RevBayes with debugging for CLion, I stumbled upon a problem. After cloning from the repo as described in [the Downloads page](https://revbayes.github.io/compile-linux), I attempted to run 
```
./build.sh -debug true
```
(alternatively, I also tried modifying `build.sh` to have `true` as the default for debug, and got the same result). This led to an error in the build. I'll attach the entire build log for reference - the error occurs around 51%. This error was reproduced for Ubuntu 20.04, Ubuntu 18.04 and Red Hat Enterprise Linux 7.6. It appears the debug build does not run well in Linux. The release build (i.e. `./build.sh` with no extra arguments) builds correctly.
[build_error.txt](https://github.com/revbayes/revbayes/files/5177253/build_error.txt)

**Comment by bredelings** [comment]
I don't have time to look further right now, but it looks like there is a compilation error:

```
In file included from /home/petrucci/revbayes/src/core/distributions/phylogenetics/substitution/PhyloCTMCSiteHomogeneousNucleotide.h:51,
                 from /home/petrucci/revbayes/src/revlanguage/distributions/phylogenetics/character/Dist_phyloCTMC.cpp:9:
/usr/lib/gcc/x86_64-linux-gnu/9/include/pmmintrin.h: In member function ‘void RevBayesCore::PhyloCTMCSiteHomogeneousNucleotide<charType>::computeInternalNodeLikelihood(const RevBayesCore::TopologyNode&, size_t, size_t, size_t, size_t) [with charType = RevBayesCore::DnaState]’:
/usr/lib/gcc/x86_64-linux-gnu/9/include/pmmintrin.h:86:1: error: inlining failed in call to always_inline ‘__m128d _mm_hadd_pd(__m128d, __m128d)’: target specific option mismatch
   86 | _mm_hadd_pd (__m128d __X, __m128d __Y)
      | ^~~~~~~~~~~
In file included from /home/petrucci/revbayes/src/revlanguage/distributions/phylogenetics/character/Dist_phyloCTMC.cpp:9:
/home/petrucci/revbayes/src/core/distributions/phylogenetics/substitution/PhyloCTMCSiteHomogeneousNucleotide.h:483:37: note: called from here
  483 |             __m128d gt = _mm_hadd_pd(g_acgt,t_acgt);
```

**Comment by brpetrucci** [comment]
Yes, those are the errors I referenced (51%). Those errors were reproduced in Ubuntu 20.04 and Red Hat Enterprise Linux 7.6. Also reproduced both in gcc 9 (as is the example you pasted) and gcc 7. The same build runs normally on Mac.

**Comment by bredelings** [comment]
It looks like `target specific option mismatch` means that the `mm_hadd_pd` instruction isn't available on the CPU that gcc is compiling for.  This is an SSE3 instruction, so GCC may need the flag `-msse3` or something like `-march=native`.

I suspect you could do `export CXXFLAGS=-march=native` to set the CXXFLAGS environment variable before running the build.  However, I wonder if this error is occurring because you already have CXXFLAGS defined to something inappropriate.

**Comment by brpetrucci** [comment]
I'll try that next time I have a chance! Since this is happening in every (3 for now, running on a fourth machine now) Linux we tried, it might be a problem worth looking into fixing on RB itself since it's not machine-specific, though

**Comment by bredelings** [comment]
Does this fix it?
https://github.com/revbayes/revbayes/pull/107

**Comment by brpetrucci** [comment]
It seems someone needs to review it to be merged, I'll try it when it is! Meanwhile we tried on a different Ubuntu machine and got the same problem. I'll try your fix soon

**Comment by bredelings** [comment]
I merged it into the `development` branch.

In the normal git workflow, you would check out the branch corresponding to the pull request, and then try that, and then we would merge it _after_ we know it helps.

**Comment by brpetrucci** [comment]
Fixed the issue both on Ubuntu 18.04 and 20.04. Seems like it worked. I could check out the branch, just didn't want to do it on a Saturday, but in any case I'm glad it worked.
